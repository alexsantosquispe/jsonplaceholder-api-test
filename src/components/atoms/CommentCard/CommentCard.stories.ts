import type { Meta, StoryObj } from '@storybook/react';

import { CommentCard } from './CommentCard';

const meta = {
  title: 'Atoms/CommentCard',
  component: CommentCard,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof CommentCard>;

export default meta;

type Story = StoryObj<typeof CommentCard>;

export const Default: Story = {
  args: {
    name: 'Anthony Stark',
    email: 'tony.stark@gmail.com',
    body: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
  }
};
