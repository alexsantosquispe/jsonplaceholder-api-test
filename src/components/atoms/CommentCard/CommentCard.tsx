import { ChatIcon, HeartIcon, AvatarIcon } from '../../../icons';
import { PostComment } from '../../../types';

type CommentCardProps = Omit<PostComment, 'id' | 'postId'>;

export const CommentCard = ({ name, email, body }: CommentCardProps) => {
  return (
    <div className="flex flex-col gap-4 text-sm">
      <div className="flex gap-2">
        <div className="h-10 w-10">
          <AvatarIcon />
        </div>

        <div className="flex flex-col gap-4">
          <div className="flex w-fit flex-col justify-center">
            <span className="text-sm font-light capitalize md:text-base">
              {name}
            </span>
            <span className="text-xs font-normal lowercase">{email}</span>
          </div>

          <div className="flex w-full flex-col gap-2">
            <p className="text-primary/65 text-xs leading-6 font-normal md:text-sm dark:text-white/65">
              {body}.
            </p>

            <div className="flex space-x-8 text-xs text-gray-400 dark:text-white/50">
              <div className="flex items-center gap-1">
                <HeartIcon className="cursor-pointer" />
                {/* This length is only used to simulate the counter */}
                {`${name.length}`}
              </div>
              <div className="flex items-center gap-1">
                <ChatIcon className="cursor-pointer" />
                {/* This length is only used to simulate the counter */}
                {`${email.length}`}
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};
