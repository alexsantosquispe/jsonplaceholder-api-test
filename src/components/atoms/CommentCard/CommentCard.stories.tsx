import type { Meta, StoryObj } from '@storybook/react';

import { CommentCard } from './CommentCard';
import { Wrapper } from '../../../utils/stories-utils';

const meta = {
  title: 'Atoms/CommentCard',
  component: CommentCard,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof CommentCard>;

export default meta;

type Story = StoryObj<typeof CommentCard>;

export const Default: Story = () => {
  return (
    <Wrapper>
      <CommentCard
        name="Anthony Stark"
        email="tony.stark@gmail.com"
        body="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
      />
    </Wrapper>
  );
};

Default.args = {};
