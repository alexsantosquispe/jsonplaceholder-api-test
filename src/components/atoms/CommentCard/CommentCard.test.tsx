import { render } from '@testing-library/react';

import { CommentCard } from './CommentCard';

describe('CommentCard', () => {
  it('should render the component correctly', () => {
    const component = render(
      <CommentCard
        name="Anthony Stark"
        email="tony.stark@gmail.com"
        body="Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi nemo ut magnam vero eum, sunt dolorem quibusdam eligendi? Pariatur, sint? Eaque nihil, vel modi ad qui voluptatibus error enim consequuntur"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
