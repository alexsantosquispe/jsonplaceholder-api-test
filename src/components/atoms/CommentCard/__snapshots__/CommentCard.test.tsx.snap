// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col gap-4 text-sm"
      >
        <div
          class="flex gap-2"
        >
          <div
            class="h-10 w-10"
          >
            <svg
              class="size-10 text-gray-200 dark:text-white/20"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M18.685 19.097A9.723 9.723 0 0 0 21.75 12c0-5.385-4.365-9.75-9.75-9.75S2.25 6.615 2.25 12a9.723 9.723 0 0 0 3.065 7.097A9.716 9.716 0 0 0 12 21.75a9.716 9.716 0 0 0 6.685-2.653Zm-12.54-1.285A7.486 7.486 0 0 1 12 15a7.486 7.486 0 0 1 5.855 2.812A8.224 8.224 0 0 1 12 20.25a8.224 8.224 0 0 1-5.855-2.438ZM15.75 9a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
          <div
            class="flex flex-col gap-4"
          >
            <div
              class="flex w-fit flex-col justify-center"
            >
              <span
                class="text-sm font-light capitalize md:text-base"
              >
                Anthony Stark
              </span>
              <span
                class="text-xs font-normal lowercase"
              >
                tony.stark@gmail.com
              </span>
            </div>
            <div
              class="flex w-full flex-col gap-2"
            >
              <p
                class="text-primary/65 text-xs leading-6 font-normal md:text-sm dark:text-white/65"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi nemo ut magnam vero eum, sunt dolorem quibusdam eligendi? Pariatur, sint? Eaque nihil, vel modi ad qui voluptatibus error enim consequuntur
                .
              </p>
              <div
                class="flex space-x-8 text-xs text-gray-400 dark:text-white/50"
              >
                <div
                  class="flex items-center gap-1"
                >
                  <svg
                    class="size-5 cursor-pointer"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  13
                </div>
                <div
                  class="flex items-center gap-1"
                >
                  <svg
                    class="size-5 cursor-pointer"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M2.25 12.76c0 1.6 1.123 2.994 2.707 3.227 1.068.157 2.148.279 3.238.364.466.037.893.281 1.153.671L12 21l2.652-3.978c.26-.39.687-.634 1.153-.67 1.09-.086 2.17-.208 3.238-.365 1.584-.233 2.707-1.626 2.707-3.228V6.741c0-1.602-1.123-2.995-2.707-3.228A48.394 48.394 0 0 0 12 3c-2.392 0-4.744.175-7.043.513C3.373 3.746 2.25 5.14 2.25 6.741v6.018Z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  20
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col gap-4 text-sm"
    >
      <div
        class="flex gap-2"
      >
        <div
          class="h-10 w-10"
        >
          <svg
            class="size-10 text-gray-200 dark:text-white/20"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M18.685 19.097A9.723 9.723 0 0 0 21.75 12c0-5.385-4.365-9.75-9.75-9.75S2.25 6.615 2.25 12a9.723 9.723 0 0 0 3.065 7.097A9.716 9.716 0 0 0 12 21.75a9.716 9.716 0 0 0 6.685-2.653Zm-12.54-1.285A7.486 7.486 0 0 1 12 15a7.486 7.486 0 0 1 5.855 2.812A8.224 8.224 0 0 1 12 20.25a8.224 8.224 0 0 1-5.855-2.438ZM15.75 9a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <div
          class="flex flex-col gap-4"
        >
          <div
            class="flex w-fit flex-col justify-center"
          >
            <span
              class="text-sm font-light capitalize md:text-base"
            >
              Anthony Stark
            </span>
            <span
              class="text-xs font-normal lowercase"
            >
              tony.stark@gmail.com
            </span>
          </div>
          <div
            class="flex w-full flex-col gap-2"
          >
            <p
              class="text-primary/65 text-xs leading-6 font-normal md:text-sm dark:text-white/65"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Quasi nemo ut magnam vero eum, sunt dolorem quibusdam eligendi? Pariatur, sint? Eaque nihil, vel modi ad qui voluptatibus error enim consequuntur
              .
            </p>
            <div
              class="flex space-x-8 text-xs text-gray-400 dark:text-white/50"
            >
              <div
                class="flex items-center gap-1"
              >
                <svg
                  class="size-5 cursor-pointer"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M21 8.25c0-2.485-2.099-4.5-4.688-4.5-1.935 0-3.597 1.126-4.312 2.733-.715-1.607-2.377-2.733-4.313-2.733C5.1 3.75 3 5.765 3 8.25c0 7.22 9 12 9 12s9-4.78 9-12Z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                13
              </div>
              <div
                class="flex items-center gap-1"
              >
                <svg
                  class="size-5 cursor-pointer"
                  fill="none"
                  stroke="currentColor"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2.25 12.76c0 1.6 1.123 2.994 2.707 3.227 1.068.157 2.148.279 3.238.364.466.037.893.281 1.153.671L12 21l2.652-3.978c.26-.39.687-.634 1.153-.67 1.09-.086 2.17-.208 3.238-.365 1.584-.233 2.707-1.626 2.707-3.228V6.741c0-1.602-1.123-2.995-2.707-3.228A48.394 48.394 0 0 0 12 3c-2.392 0-4.744.175-7.043.513C3.373 3.746 2.25 5.14 2.25 6.741v6.018Z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                  />
                </svg>
                20
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
